hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -1128,3 +1128,47 @@ def hpat_pandas_series_le_impl(self, other):
         return hpat_pandas_series_le_impl
 
     raise TypingError('{} The object must be a pandas.series and argument must be a number. Given: {} and other: {}'.format(_func_name, self, other))
+
+   
+@overload_method(SeriesType, 'quantile')
+def hpat_pandas_series_quantile(self, q=0.5, interpolation='linear'):
+    """
+    Pandas Series method :meth:`pandas.Series.quantile` implementation.
+
+    .. only:: developer
+
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_number_quantile
+
+    Parameters
+    -----------
+    q : :obj: float or array-like object
+              default 0.5
+    interpolation: ‘linear’, ‘lower’, ‘higher’, ‘midpoint’, ‘nearest’
+                 *unsupported*
+
+    Returns
+    -------
+    :obj:`pandas.Series` or float
+    """
+
+    _func_name = 'Method quantile().'
+
+    supported_q_types = (float, types.Float, types.Omitted)
+
+    if not isinstance(self, SeriesType):
+        raise TypingError(
+            '{} The object must be a pandas.series. Given self: {}'.format(_func_name, self))
+
+    if not isinstance(interpolation, types.Omitted) and interpolation is not 'linear':
+        raise TypingError(
+            '{} Unsupported parameters. Given interpolation: {}'.format(_func_name, interpolation))
+
+    if type(q) not in supported_q_types:
+        raise TypingError(
+            '{} The parameter must be float. Given type q: {}'.format(_func_name, type(q)))
+
+    def hpat_pandas_series_quantile_impl(self, q=0.5, interpolation='linear'):
+        return np.quantile(self._data, q)
```

Comment:
Do we need to use Numpy here explicitly? Could we use `self._data.quantile()` instead?

Reply:
I'm not sure, because Numba just overloads np.quantile. There is no such method for array type in Numba

Reply:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.quantile.html
This is new method presented since v1.15

Reply:
I mean we can only call as np.quantile(self._data, q), like in examples by provided link